python/TestHarness/schedulers/RunParallel.py

```diff
@@ -0,0 +1,130 @@
+from timeit import default_timer as clock
+
+from Tester import Tester
+from Scheduler import Scheduler
+from MooseObject import MooseObject
+from util import *
+import platform, time
+
+import os
+
+## This class provides an interface to run commands in parallel
+#
+# To use this class, call the .run() method with the tester options
+class RunParallel(Scheduler):
+    @staticmethod
+    def validParams():
+        params = Scheduler.validParams()
+        params.addRequiredParam('scheduler',    'RunParallel', "The name of this scheduler")
+
+        return params
+
+    def __init__(self, harness, params):
+        Scheduler.__init__(self, harness, params)
+
+    ## Run the test!
+    ## Note: use thread_lock where necessary
+    #
+    #  with thread_lock:
+    #      do protected stuff
+    #
+    # The run method should be blocking until the test has completed _and_ the results have been
+    # processed (processResults()). When we return from this method, this test will be considered
+    # finished and added to the status pool for printing out the final results.
+    def run(self, tester, testers, thread_lock, options):
+        # Ask the tester if its allowed to run on this machine
+        if tester.checkRunnableBase(options):
+
+            # Ask if this test is allowed to run based on other test's circumstances
+            status_check = StatusDependency(tester, testers, options)
```

Comment:
I suppose it doesn't have to be a new status dependency object. I could shimmy it into the TestStatus class perhaps. Or just a couple of methods not needing a class. I am surely not using it correctly by having the need to initialize it with a constructor. Overkill. Its only instanced once and thrown away...